internal/oci/uvm.go

```diff
@@ -313,6 +314,49 @@ func parseAnnotationsString(a map[string]string, key string, def string) string
 	return def
 }
 
+// handleAnnotationKernelDirectBoot handles parsing annotationKernelDirectBoot and setting
+// implied annotations from the result.
+func handleAnnotationKernelDirectBoot(ctx context.Context, a map[string]string, lopts *uvm.OptionsLCOW) {
+	lopts.KernelDirect = parseAnnotationsBool(ctx, a, annotationKernelDirectBoot, lopts.KernelDirect)
+	if !lopts.KernelDirect {
+		lopts.KernelFile = uvm.KernelFile
+	}
+}
+
+// handleAnnotationPreferredRootFSType handles parsing annotationPreferredRootFSType and setting
+// implied annotations from the result
+func handleAnnotationPreferredRootFSType(ctx context.Context, a map[string]string, lopts *uvm.OptionsLCOW) {
+	lopts.PreferredRootFSType = parseAnnotationsPreferredRootFSType(ctx, a, annotationPreferredRootFSType, lopts.PreferredRootFSType)
+	switch lopts.PreferredRootFSType {
+	case uvm.PreferredRootFSTypeInitRd:
+		lopts.RootFSFile = uvm.InitrdFile
+	case uvm.PreferredRootFSTypeVHD:
+		lopts.RootFSFile = uvm.VhdFile
+	}
+}
+
+// handleAnnotationFullyPhysicallyBacked handles parsing annotationFullyPhysicallyBacked and setting
+// implied annotations from the result. For both LCOW and WCOW options.
+func handleAnnotationFullyPhysicallyBacked(ctx context.Context, a map[string]string, opts interface{}) {
+	switch opts.(type) {
```

Comment:
Forgot to put this but you can avoid doing the type conversion steps in the case blocks by doing 
```go 
switch options := opts.(type) {
case *uvm.OptionsLCOW:
         options.FullyPhysicallyBacked = .............
case *uvm.OptionsWCOW:
         options.FullyPhysicallyBacked = ........
}
```

Reply:
Would you prefer that?